src/components/swap/swap.tsx

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+
+import {observer} from 'mobx-react';
+import {View} from 'react-native';
+
+import {createTheme} from '@app/helpers';
+import {useSumAmount} from '@app/hooks';
+import {I18N} from '@app/i18n';
+import {Balance} from '@app/services/balance';
+import {
+  SushiPoolEstimateResponse,
+  SushiPoolResponse,
+} from '@app/services/indexer';
+import {IContract} from '@app/types';
+import {STRINGS} from '@app/variables/common';
+
+import {
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  First,
+  Spacer,
+  Text,
+  TextField,
+  TextVariant,
+} from '../ui';
+import {Placeholder} from '../ui/placeholder';
+
+export interface SwapProps {
+  poolData: SushiPoolResponse;
+  estimateData: SushiPoolEstimateResponse | null;
+  tokenIn: IContract;
+  tokenOut: IContract;
+  amountsIn: ReturnType<typeof useSumAmount>;
+  amountsOut: ReturnType<typeof useSumAmount>;
+  isEstimating: boolean;
+  isSwapInProgress: boolean;
+  isApproveInProgress: boolean;
+  estimate(): Promise<void>;
+  onPressChangeTokenIn(): Promise<void>;
+  onPressChangeTokenOut(): Promise<void>;
+  onPressSwap(): Promise<void>;
+  onPressApprove(): Promise<void>;
+}
+
+export const Swap = observer(
+  ({
+    amountsIn,
+    amountsOut,
+    estimateData,
+    isEstimating,
+    // poolData,
+    tokenIn,
+    tokenOut,
+    isApproveInProgress,
+    isSwapInProgress,
+    estimate,
+    onPressChangeTokenIn,
+    onPressChangeTokenOut,
+    onPressApprove,
+    onPressSwap,
+  }: SwapProps) => {
+    const needApproveTokenIn = estimateData?.need_approve;
+    const isLoading = isEstimating;
```

Comment:
https://eslint.org/blog/2024/04/eslint-v9.0.0-released/#new-rule%3A-no-useless-assignment

😄

Reply:
You can rename variable in props, if you don't like